swarm/network/kademlia_test.go

```diff
@@ -566,79 +566,73 @@ func newTestDiscoveryPeer(addr pot.Address, kad *Kademlia) *Peer {
 // when neighbourhood depth is changed.
 func TestKademlia_SubscribeToNeighbourhoodDepthChange(t *testing.T) {
 
-	testSignal := func(t *testing.T, k *testKademlia, prevDepth int, c <-chan struct{}) (newDepth int) {
+	testSignal := func(t *testing.T, k *testKademlia, prevDepth int) chan int {
 		t.Helper()
-
-		select {
-		case _, ok := <-c:
-			if !ok {
-				t.Error("closed signal channel")
-			}
-			newDepth = k.NeighbourhoodDepth()
-			if prevDepth == newDepth {
-				t.Error("depth not changed")
+		doneC := make(chan int)
+		go func() {
+			d := 0
```

Comment:
Maybe use `var d int`, instead of `d := 0`, because you really don't care about the initial value, but rather care about the scope of `d`. The initialisation is done in the `for` loop.